packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -587,3 +595,52 @@ export class TelemetryNullLogger implements ITelemetryLogger {
     public sendErrorEvent(event: ITelemetryErrorEvent, error?: any): void {}
     public sendPerformanceEvent(event: ITelemetryPerformanceEvent, error?: any): void {}
 }
+
+/**
+ * Takes in an event object, and converts all of its values to a basePropertyType.
+ * In the case of an invalid property type, the value will be converted to an error string.
+ * @param event - Event with fields you want to stringify.
+ */
+function convertToBaseEvent({ category, eventName, ...props }: ITelemetryEventExt): ITelemetryBaseEvent {
+    const newEvent: ITelemetryBaseEvent = { category, eventName };
+    for (const key of Object.keys(props)) {
+        newEvent[key] = convertToBasePropertyType(props[key]);
+    }
+    return newEvent;
+}
+
+/**
+ * Takes in value, and does one of 4 things.
+ * if value is of primitive type - returns the original value.
+ * If the value is an array of primitives - returns a stringified version of the array.
+ * If the value is an object of type ITaggedTelemetryPropertyType - returns the object
+ * with its values recursively converted to base property Type.
+ * If none of these cases are reached - returns an error string
+ * @param x - value passed in to convert to a base property type
+ */
+function convertToBasePropertyType(
+    x: TelemetryEventPropertyTypeExt | ITaggedTelemetryPropertyTypeExt,
+): TelemetryEventPropertyType | ITaggedTelemetryPropertyType {
+    switch (typeof x) {
+        case "string":
+        case "number":
+        case "boolean":
+        case "undefined":
+            return x;
+        case "object":
+            if (Array.isArray(x)) {
+                // We know that the array type is
+                return JSON.stringify(x);
+            }
+            return {
+                value: JSON.stringify(
```

Comment:
Try this test case:  `convertToBasePropertyType({value: true, tag: foo })`.  I don't think you want the `JSON.stringify`.